### PR TITLE
Two pane skip todo interface

### PR DESCRIPTION
**Pull request by adbharadwaj**
screen shot 2018-06-14 at 5 45 31 pm

### CODE REVIEW

File: orchestra/static/orchestra/todos/todo-list.directive.es6.js [hunk 1]
```diff
@@ -98,6 +104,7 @@ export default function todoList (orchestraApi) {
               return result
             }, {})
           todoList.possibleTasks = Object.values(response.data.tasks).filter(task => task.status !== 'Complete' && humanSteps.has(task.step_slug))
+          todoList.possibleRoles = ['CSM', 'Designer']
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-list.directive.es6.js]: role comes from possibleTasks :)

e.g., https://github.com/b12io/orchestra/blob/master/orchestra/static/orchestra/todos/todo-list.html#L21

File: orchestra/static/orchestra/todos/todo-list.html [hunk 2]
```diff
@@ -3,83 +3,85 @@
...
+          Checklist
         </h3>
       </div>
     </div>
     <div class="row section-body" ng-if="todoList.ready">
-      <div class="col-lg-12 col-md-12 col-sm-12">
-        <form class="new-todo">
-          <div class="new-todo__box">
+      <div class="col-lg-6 col-md-6 col-sm-12">
+        <div class="checklist-pane">
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-list.html]: since everything in Orchestra is called todos, let's keep naming to todo/todolist

Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-list.html]: ok :)

File: orchestra/models/core/models.py [hunk 3]
```diff
@@ -576,6 +576,37 @@ class PayRate(PayRateMixin, models.Model):
     end_date = models.DateField(null=True, blank=True)
```
Comment (paopow) [inline orchestra/models/core/models.py:578]: Lint. :)

File: orchestra/static/orchestra/todos/todo-list.html [hunk 4]
```diff
@@ -4,82 +4,84 @@
...
-              show-checked="false"
+              show-skipped="false"
               update-todo="todoList.updateTodo"
+              toggle-skip-todo="todoList.toggleSkipTodo"
               steps="todoList.steps"
               task-slugs="todoList.taskSlugs"
-            ></todo-checklist>
-            <todo-checklist
-              title="Completed"
+          ></todo-checklist>
```
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-list.html:67]: Shouldn't there be two todo-checklist's: 1) Todos and 2) Completed?

Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-list.html:67]: @paopow That was the case in the previous implementation. I thought it was redundant to show checked (completed) items in a dedicated 'Completed' list.

File: orchestra/models/core/models.py (hunk 3 above)
Comment (adbharadwaj) [reply orchestra/models/core/models.py:578]: Yup :)

File: orchestra/static/orchestra/todos/todo-list.html (hunk 4 above)
Comment (paopow) [reply orchestra/static/orchestra/todos/todo-list.html:67]: Hm... One reason we had a separated completed list was to preserve the CSMs' sanity. It probably doesn't make that much difference. Let's see what @marcua think.

Comment (marcua) [reply orchestra/static/orchestra/todos/todo-list.html:67]: I'm cool leaving it out until we hear feedback that it's too overwhelming to have incomplete items there.

I'm also cool adding it now---no strong intuition from expert interviews or anything.

File: orchestra/models/core/models.py [hunk 5]
```diff
@@ -607,6 +646,13 @@ class Meta:
     completed = models.BooleanField(default=False)
     start_by_datetime = models.DateTimeField(null=True, blank=True)
     due_datetime = models.DateTimeField(null=True, blank=True)
+    skipped = models.BooleanField(default=False)
```
Comment (marcua) [inline orchestra/models/core/models.py]: you can remove skipped so we don't have derived data :)

File: orchestra/json_schemas/todos.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import jsl
+
+
+class TodoSchema(jsl.Document):
+    id = jsl.IntField(required=True)
+    description = jsl.StringField(required=True)
+    items = jsl.DocumentField('TodoSchema', required=True)
```
Comment (marcua) [inline orchestra/json_schemas/todos.py]: the children won't be sorted if you do this, right?

you might end up needing an ArrayList of (slug, Todo) pairs?

File: orchestra/static/orchestra/todos/todo-list.html (hunk 4 above)
Comment (paopow) [reply orchestra/static/orchestra/todos/todo-list.html:67]: Let's not have it then and see whether people want it back.

File: orchestra/models/core/models.py [hunk 7]
```diff
@@ -576,6 +576,35 @@ class PayRate(PayRateMixin, models.Model):
...
+    """
+    class Meta:
+        app_label = 'orchestra'
+
+    slug = models.CharField(max_length=200, unique=True)
+    name = models.CharField(max_length=200)
+    description = models.TextField()
+    creator = models.ForeignKey(
+        Worker, null=True, blank=True,
+        related_name='creator', on_delete=models.CASCADE)
```
Comment (marcua) [inline orchestra/models/core/models.py]: SET_NULL?

File: orchestra/models/core/models.py [hunk 8]
```diff
@@ -590,7 +619,17 @@ class Todo(TodoMixin, BaseModel):
...
         due_datetime (datetime.datetime):
-            The time the todo is due.
+            The time the todo is due
+        skipped_datetime (datetime.datetime):
+            The time the todo was skipped
+        parent_todo (orchestra.models.Todo):
+            The parent todo item
+        template (orchestra.models.TodoListTemplate)
+            The template the todo is based on
+        log (str)
```
Comment (marcua) [inline orchestra/models/core/models.py]: activity_log

File: orchestra/models/core/models.py [hunk 9]
```diff
@@ -607,6 +646,12 @@ class Meta:
     completed = models.BooleanField(default=False)
     start_by_datetime = models.DateTimeField(null=True, blank=True)
     due_datetime = models.DateTimeField(null=True, blank=True)
+    skipped_datetime = models.DateTimeField(null=True, blank=True)
+    parent_todo = models.ForeignKey(
+        'self', null=True, related_name='parent')
+    template = models.ForeignKey(
```
Comment (marcua) [inline orchestra/models/core/models.py]: on_delete SET_NULL

File: orchestra/models/core/models.py [hunk 10]
```diff
@@ -607,6 +646,12 @@ class Meta:
     completed = models.BooleanField(default=False)
     start_by_datetime = models.DateTimeField(null=True, blank=True)
     due_datetime = models.DateTimeField(null=True, blank=True)
+    skipped_datetime = models.DateTimeField(null=True, blank=True)
+    parent_todo = models.ForeignKey(
```
Comment (marcua) [inline orchestra/models/core/models.py:650]: on_delete CASCADE

**Review by marcua — APPROVED** [review]

**Pull request merged**